src/Query/Builder.php

```diff
@@ -426,7 +427,7 @@ public function toMql(): array
 
             // Add custom query options
             if (count($this->options)) {
-                $options = array_merge($options, $this->options);
+                $options = array_replace($options, $this->options);
```

Comment:
There's likely no concern for numeric keys here, but as we discussed it's more expressive to use `array_replace()` when merging options as well.